File: sylee723/1주차/PGS_더_맵게.java
import java.util.*; 

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int i = 0; i < scoville.length; i++) {
            pq.add(scoville[i]);
        }
        
        int answer = 0;
        int mix;
        while (true) {
            if (pq.peek() >= K)
                break;
            
            if (pq.size() < 2) {
                answer = -1;
                break;
            }
            
            mix = pq.poll() + (pq.poll() * 2);
            pq.add(mix);
            answer++;
        }
        
        return answer;
    }
}

File: daehwa_park/3주차/PGS_더_맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    
    heapq.heapify(scoville)
    
    while True:
        first = heapq.heappop(scoville)
        if first < K and len(scoville) > 0:
            second = heapq.heappop(scoville)
            mix = first + (second * 2)
            heapq.heappush(scoville, mix)
            answer += 1
        elif first < K and len(scoville) == 0:
            break
        else:
            return answer
        
    return -1